observability/src/main/java/io/grpc/observability/interceptors/LogHelper.java

```diff
@@ -0,0 +1,436 @@
+        T message,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+      checkArgument(
+          eventType == EventType.GRPC_CALL_REQUEST_MESSAGE
+              || eventType == EventType.GRPC_CALL_RESPONSE_MESSAGE,
+          "event type must correspond to client message or server message");
+      checkNotNull(message, "message");
+
+      // TODO(dnvindhya): Convert message to bystestring
+      // byte[] messageArray = (byte[])message;
+      // int messageLength = messageArray.length;
+      // ByteString messageData =
+      // ByteString.copyFrom((byte[]) message, 0, ((byte[]) message).length);
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(eventType)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINEST, "Writing RPC message to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+
+    /**
+     * Logs half close.
+     */
+    void logHalfClose(
+        long seqId,
+        String serviceName,
+        String methodName,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(EventType.GRPC_CALL_HALF_CLOSE)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINEST, "Writing Half Close event to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+
+    /**
+     * Logs cancellation.
+     */
+    void logCancel(
+        long seqId,
+        String serviceName,
+        String methodName,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(EventType.GRPC_CALL_CANCEL)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINEST, "Writing Cancel event to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+  }
+
+  // Copied from internal
+  static final class ByteArrayMarshaller implements Marshaller<byte[]> {
```

Comment:
- I notice this is only used in a test right now (`InternalLoggingChannelInterceptorTest`) . What's the purpose of this class?
- if this was because of the binary logger then note that we don't do "binary" logging of various rpc parts so we don't really need the binary marshaller (and its counterpart unmarshaller) and we can just use the byteString for message payload logging.

Reply:
This is required by tests for creating a MethodDescriptor which also required setting a Request and Response Marshaller. So moved it to `LogHelperTest` class